core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -595,24 +595,23 @@ public void setDefaultTTL(int defaultTTL) {
 
         this.updateMetricExpiration(metric);
 
-        return dataAccess.addTags(metric, tags).mergeWith(dataAccess.insertIntoMetricsTagsIndex(metric, tags))
-                .toList().map(l -> null);
+        return dataAccess.addTags(metric, tags).map(l -> null);
     }
 
     @Override
     public Observable<Void> deleteTags(Metric<?> metric, Set<String> tags) {
         return getMetricTags(metric.getMetricId())
                 .map(loadedTags -> {
-                    loadedTags.keySet().retainAll(tags);
+                    if (tags != null) {
```

Comment:
Yes, but my point was more about: why would we call "deleteTags" with null tags? And if really we want to do so, then we could just return an empty observable at the beginning of "deleteTags" and thus avoid an unnecessary call to "getMetricsTags".